### PR TITLE
feat(download): enable custom file name for downloads

### PR DESCRIPTION
**Pull request by SyntaxRules**
Adds file name option for #102 

Tested via the yarn server and changing the config for the test chart.

### CODE REVIEW

File: test-vl.html [hunk 1]
```diff
@@ -51,7 +51,8 @@
 
       vegaEmbed("#vis", spec, {
         config: config,
-        tooltip: { theme: "dark" }
+        tooltip: { theme: "dark" },
+        downloadFileName: "my_chart"
```
Comment (domoritz) [inline test-vl.html]: Let's not add this here.

**Comment by SyntaxRules** [comment]
@domoritz Unit test's added and review comments have been implemented.

**Comment by domoritz** [comment]
This is great. Thank you for the code and tests.

**Comment by MariaNattestad** [comment]
Thanks for this!
For Altair users, here's how you can pass this parameter through to your saved HTML file:
```
my_chart.save("my_chart.html", embed_options={"downloadFileName": "FILENAME"})
```
That FILENAME in turn ends up here in the HTML, which you could also add yourself if using vega-embed without Altair:
```
var embedOpt = {"mode": "vega-lite", "downloadFileName": "FILENAME"};
...
 vegaEmbed("#vis", spec, embedOpt)
```
It took a bit of digging, so I wanted to leave this here in case it helps someone else using Altair or vega-embed.

**Comment by domoritz** [comment]
Thank you for posting this trick! 

Note that you can even pass the embed options as part of the spec in the `usermeta` property.

**Pull request merged**